### DISCUSSION TITLE
PWM output and capture on the same timer does not work on STM32F411

### DISCUSSION THREAD

**Discussion by fcgdam** [zephyrproject-rtos/zephyr#79370] [Q&A]
Not sure if this is a bug, but I'm having a problem with generating a PWM signal and capturing it on the same Timer.

So the base DTS file is:

```
/ {
    pwm_loopback_0 {
        compatible = "test-pwm-loopback";
        /* first index must be a 32-Bit timer */
        pwms = <&pwm1 1 256 PWM_POLARITY_NORMAL>,
               <&pwm1 2 256 PWM_POLARITY_NORMAL>;
    };  
};

&timers1 {
    status = "okay";
    st,prescaler = <2>;
    pwm1: pwm {
        status = "okay";
        pinctrl-0 = <&tim1_ch1_pa8 &tim1_ch2_pa9>;  /* Define two PWM channels. Check CubeMX for the correct pins. */
        pinctrl-names = "default";
    };
};

&timers2 {
   status = "okay";
   st,prescaler = <2>;
    pwm2: pwm {
        status = "okay";
        pinctrl-0 = <&tim2_ch1_pa0 &tim2_ch2_pa1>;
        pinctrl-names = "default";
    };
};
```
So it seems with this configuration if the TIM1 (PA8) output channel 1 is connected to channel 2 (PA9), PA9 is pulling the signal low, and so there is no output and so, no capture. The capture fails with -11 wish is EAGAIN or timeout on capture.

I can see on the oscilloscope that the signal is grounded. If I remove the connection between PA8 and PA9, then PWM signal occurs correctly on PA8. 

If however I change the capture channel to TIM2, channel 1, PA0, everything works fine. 

```
....
<&pwm2 1 256 PWM_POLARITY_NORMAL>;
....
```
Some notes:
 - PWM signal can be generated either on PA8 or PA9 (CH1 or CH2) without issues on Timer 1.
 - Same is true as for Timer 2
 - The issue only happens if output and input are using the same timer.

Any ideas?

**Comment by fcgdam** [comment]
Another quick note for future reference:

- Having an output PWM signal and an input PWM on the same timer works, as long that the input does not come from the same timer, at least on this board (Weact F411ce).
- So having PWM generation at TIM1 CH1 and TIM2 CH1 and PWM input on TIM2 CH2, works as long that the input is connected to TIM1 and not TIM2.

**Comment by oor-brad** [comment]
Hey @fcgdam , any updates on this thread?
I'm seeing similar where I'm trying to provision two PWMs on the same timer, one input and one output. What I see is a bit different though; I configure the output first and confirm all is well on my scope, but once I configure the input, the output halts. 

In my case, I was wondering if there is a priority conflict, since both are operating in the same thread, but your issue sounds very similar and thought it worth a comment.

**Reply by fcgdam** [reply]
Hi:

Sorry, but no. I've moved forward and didn't test it any further. Nevertheless I did not test it with the most recent releases or repository head, so it might have change, but it seems it didn't.
